lib/signed-xml.js

```diff
@@ -411,7 +439,7 @@ SignedXml.prototype.validateSignatureValue = function(doc) {
 SignedXml.prototype.calculateSignatureValue = function(doc) {
   var signedInfoCanon = this.getCanonSignedInfoXml(doc)
   var signer = this.findSignatureAlgorithm(this.signatureAlgorithm)
-  this.signatureValue = signer.getSignature(signedInfoCanon, this.signingKey)
+  return this.signatureValue = signer.getSignature(signedInfoCanon, this.signingKey)
```

Comment:
Having read the rest of this PR,  I have to ask - why break convention and potentially return a Promise here? Wouldn't it be more consistent to have a callback in the argument?